agent/worker.go

```diff
@@ -484,3 +489,36 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		}
 	}
 }
+
+// Leave removes all assignments
+func (w *worker) Leave(ctx context.Context) error {
+	var err error
+	w.Close()
+
+	if err = w.Assign(ctx, nil); err != nil {
+		return err
+	}
+
+	// wait and confirm tasks are removed from the node
+	for retries := 0; retries < 3; retries++ {
```

Comment:
This should definitely not be a busy wait with a 3 retry loop. We should register a channel that gives us changes on any `Task` status, and then inspect if there are any Tasks left, until there are none.